src/ImageClassifier/index.js

```diff
@@ -82,43 +82,39 @@ class ImageClassifier {
   }
 
   async loadModelFrom(path = null) {
-    fetch(path)
-      .then(r => r.json())
-      .then((r) => {
-        if (r.ml5Specs) {
-          this.mapStringToIndex = r.ml5Specs.mapStringToIndex;
-        }
-      })
-      // When loading model generated by Teachable Machine 2.0, the r.ml5Specs is missing,
-      // which is causing imageClassifier failing to display lables.
-      // In this case, labels are stored in path/./metadata.json
-      // Therefore, I'm fetching the metadata and feeding the labels into this.mapStringToIndex
-      // by Yang Yang, yy2473@nyu.edu, Oct 2, 2019
-      .then(() => {
-        if (this.mapStringToIndex.length === 0) {
-          const split = path.split("/");
-          const prefix = split.slice(0, split.length - 1).join("/");
-          const metadataUrl = `${prefix}/metadata.json`;
-          fetch(metadataUrl)
-            .then((res) => {
-              if (!res.ok) {
-                console.log("Tried to fetch metadata.json, but it seems to be missing.");
-                throw Error(res.statusText);
-              }
-              return res;
-            })
-            .then(metadataJson => metadataJson.json())
-            .then((metadataJson) => {
-              if (metadataJson.labels) {
-                this.mapStringToIndex = metadataJson.labels;
-              }
-            })
-            .catch(() => console.log("Error when loading metadata.json"));
+    try {
+
+      let result;
+      let data;
+      if(path !== null){
```

Comment:
One somewhat irrelevant question on this: It looks like the previous version of this function would do straight to the catch block if a null path was passed into fetch, which at face value makes sense to me. Are there other reasons that we might want to avoid returning early or allow this function to take a null path?

Reply:
No such thing as irrelevant questions!
I think what was happening in the last code was 2 things:
1. The first fetch() promise chain was running but not finishing before calling the `tf.loadLayersModel()` at the end. 
2. next if an error was being thrown somewhere in that promise chain, the maybe we weren't getting to load the model? Or at least when the tests were running, those errors get caught whereas in the browser, they don't prevent the model from loading eventually.

I think that might explain why the tests fail but the examples still run?

Reply:
Great point! That helps me to understand why this fixed the test issue!

How should this function respond to a null path input?

Reply:
Ah so what might be a bit confusing is that we have:

```js
async loadModel(modelUrl) {
    if (modelUrl) this.model = await this.loadModelFrom(modelUrl);
    else this.model = await this.modelToUse.load({ version: this.version, alpha: this.alpha });

    return this;
  }
```

For the pre-trained models that are baked into ml5, these would get resolved before ever getting to the `loadModelFrom()` function.

Once we get into the `loadModelFrom()` function the `null path` would need to actually get caught as an error!